### PR TITLE
Shorten CLI identifiers

### PR DESCRIPTION
**Pull request by iverberk**
* Truncate all UUID identifiers to eight characters by default
* Refactor the node identifier to an auto-generated UUID
* Created and updated tests and mocks

### CODE REVIEW

**Comment by iverberk** [comment]
Not sure what is going on with tests failing. They run without errors on my local machine. Will take a look at them tomorrow. Would be nice to know if this is happening locally to other people as well.

File: client/client.go [hunk 1]
```diff
@@ -450,6 +450,11 @@ func (c *Client) setupNode() error {
 		node = &structs.Node{}
 		c.config.Node = node
 	}
+	id, err := c.nodeID()
```
Comment (dadgar) [inline client/client.go]: can you do:
```
// Generate an ID for the node
var err error
node.id, err := c.nodeID()
if err != nil {...}
```

File: command/monitor.go [hunk 2]
```diff
@@ -60,13 +60,14 @@ type monitor struct {
...
 	state  *evalState
+	length int
 
 	sync.Mutex
 }
 
 // newMonitor returns a new monitor. The returned monitor will
 // write output information to the provided ui.
-func newMonitor(ui cli.Ui, client *api.Client) *monitor {
+func newMonitor(ui cli.Ui, client *api.Client, length int) *monitor {
```
Comment (dadgar) [inline command/monitor.go]: Can you add a comment on length here and in the struct

**Comment by dadgar** [comment]
LGTM! Sorry this is a bit tedious but can you update the documentation on these pages: https://www.nomadproject.io/docs/commands/agent-info.html

**Comment by iverberk** [comment]
Thanks for the review! I updated the code according to your comments. The website documentation has also been updated, not tedious at all, just part of maintaining a solid product.

**Comment by dadgar** [comment]
Awesome work! Thanks @iverberk

**Comment by iverberk** [comment]
Sweet, thanks! :-) Maybe also good to mention in the changelog that the node id can no longer be specified through the configuration (backwards incompatible)?

**Comment by dadgar** [comment]
Yeah I threw it under the "BACKWORDS INCOMPATIBLE" section. We will have to
put together an upgrade guide for 0.2->0.3

**Pull request merged**